sql-cli/sql_cli/configuration.py

```diff
@@ -39,42 +39,49 @@ class Config:
     airflow_home: str | None = None
     airflow_dags_folder: str | None = None
 
-    def get_filepath(self) -> Path:
+    def get_env_config_filepath(self) -> Path:
         """
-        Return configuration.yaml filepath.
+        Return environment specific configuration.yaml filepath.
 
         :returns: The path to the desired YAML configuration file
         """
         return self.project_dir / CONFIG_DIR / self.environment / CONFIG_FILENAME
 
-    def from_yaml_to_dict(self) -> dict[str, Any]:
+    def get_global_config_filepath(self) -> Path:
         """
-        Return a dict with the content of the configuration.yaml.
+        Return global configuration.yaml filepath which is shared across environments.
+
+        :return: The path to the desired global YAML configuration file
+        """
+        return self.project_dir / CONFIG_DIR / GLOBAL_CONFIG / CONFIG_FILENAME
+
+    @staticmethod
+    def from_yaml_to_dict(filepath: Path) -> dict[str, Any]:
+        """
+        Return a dict with the contents of the given configuration.yaml
+
+        :param filepath: Path of the desired configuration.yaml to read contents from.
 
         :returns: Content of the YAML configuration file as a python dictionary.
         """
-        filepath = self.get_filepath()
         with open(filepath) as fp:
             yaml_with_env = os.path.expandvars(fp.read())
             yaml_config = yaml.safe_load(yaml_with_env)
         return yaml_config
 
     def from_yaml_to_config(self) -> Config:
-        """
-        Return a Config instance with the content of the configuration.yaml.
-
-        :returns: Contents of the YAML configuration file.
-        """
-        yaml_config = self.from_yaml_to_dict()
+        """Returns a Config instance with the contents of the environment specific and global configuration.yaml"""
+        env_yaml_config = self.from_yaml_to_dict(self.get_env_config_filepath())
+        global_yaml_config = self.from_yaml_to_dict(self.get_global_config_filepath()) or {}
```

Comment:
According to the `from_yaml_to_dict` function definition it always returns a dict. So `or {}` will never be reached. Either the return type is wrong or the `or {}` can be removed. But my guess is that the return type is wrong in case the global config does not exist, correct?

Reply:
Yes, if the file content is empty, the return is None, hence was converting it to empty dictionary. But made the change to instead return empty dictionary from `from_yaml_to_dict` in case the return is `None`.